### PR TITLE
Create entity

### PR DESCRIPTION
**Pull request by firuza0029**
Entities and migrations are created

### CODE REVIEW

File: InternshipBE/DAL/Database/ApplicationDbContext.cs [hunk 1]
```diff
@@ -10,9 +16,29 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : ba
 		{
 		}
 
+		public DbSet<Feedback> Feedbacks { get; set; }
+
+		public DbSet<Evaluation> Evaluations { get; set; }
+
+		public DbSet<Skill> Skills { get; set; }
+
+        public DbSet<Role> Roles { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Database/ApplicationDbContext.cs]: remove Role entity

File: InternshipBE/DAL/Entities/Internship.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using Shared.Enums;
+using System;
+
+namespace DAL.Entities
+{
+    public class Internship
+    {
+        public int Id { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Internship.cs:9]: spaces between prop

File: InternshipBE/DAL/Entities/Candidate.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        public Stack Stack { get; set; }
+
+        public EnglishLevel EnglishLevel { get; set; }  
+
+        public bool ArePlainingToJoin { get; set; }
+
+        public DateTime RegistationDate { get; set; }
+
+        public Internship Internship { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Candidate.cs:50]: remove empty line

File: InternshipBE/DAL/Entities/Evaluation.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    public class Evaluation
+    {
+        public int Id { get; set; }
+
+        public int FeedbackId { get; set; }
+
+        public int SkillId { get; set; }
+
+        public int Value { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Evaluation.cs:14]: remove empty line

File: InternshipBE/DAL/Entities/Feedback.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        public int Id { get; set; }
+
+        public int UserCandidateId { get; set; }
+
+        public EnglishLevel EnglishLevel { get; set; }
+
+        public DateTime Date { get; set; }
+
+        public string Description { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Feedback.cs:18]: remove empty line

File: InternshipBE/DAL/Entities/Internship.cs (hunk 2 above)
Comment (romanstat) [reply InternshipBE/DAL/Entities/Internship.cs:9]: where is relationship to candidate?

File: InternshipBE/DAL/Entities/InterviewInvite.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+namespace DAL.Entities
+{
+    public class InterviewInvite
+    {
+        public int Id { get; set; }
+
+        public int UserCandidateId { get; set; }
+
+        public DateTime ContactDate { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/InterviewInvite.cs:12]: remove empty line

File: InternshipBE/DAL/Entities/Role.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace DAL.Entities
+{
+    public class Role
+    {
+        public int Id { get; set; }
+
+        public string Name { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Role.cs]: remove empty line

File: InternshipBE/DAL/Entities/Skill.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    public class Skill
+    {
+        public int Id { get; set; }
+
+        public Stack Stack { get; set; }
+
+        public string Name { get; set; }
+
+        public string IsHardSkill { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Skill.cs:16]: remove empty line

File: InternshipBE/DAL/Entities/UserCandidate.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace DAL.Entities
+{
+    public class UserCandidate
+    {
+        public int Id {  get; set; }
+        public int UserId {  get; set; }
+        public int CandidateId { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/UserCandidate.cs:13]: remove empty line, spaces between prop

File: InternshipBE/WebApi/Startup.cs [hunk 10]
```diff
@@ -27,12 +28,14 @@ public Startup(IConfiguration configuration)
...
 		public void ConfigureServices(IServiceCollection services)
 		{
+
 			services.AddDbContext<ApplicationDbContext>(options =>
 				options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
 
 			services.AddIdentity<User, IdentityRole>()
 				.AddEntityFrameworkStores<ApplicationDbContext>();
 
+			services.AddDbContext<ApplicationDbContext>(options => options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
```
Comment (romanstat) [inline InternshipBE/WebApi/Startup.cs]: line 32, AddDbContext already exists

File: InternshipBE/DAL/Database/ApplicationDbContext.cs [hunk 11]
```diff
@@ -10,9 +16,29 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : ba
...
+
+		public DbSet<Candidate> Candidates { get; set; }
+
+		public DbSet<Internship> Internships { get; set; }
+
+		public DbSet<InternshipStack> InternshipStacks { get; set; }
+
+		public DbSet<UserCandidate> UserCandidates { get; set; }
+
+
```
Comment (romanstat) [inline InternshipBE/DAL/Database/ApplicationDbContext.cs]: Where is identity user?

File: InternshipBE/DAL/Database/ApplicationDbContext.cs [hunk 12]
```diff
@@ -10,9 +16,29 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : ba
...
 		protected override void OnModelCreating(ModelBuilder builder)
 		{
 			base.OnModelCreating(builder);
 		}
+
```
Comment (romanstat) [inline InternshipBE/DAL/Database/ApplicationDbContext.cs]: remove empty line

File: InternshipBE/DAL/Database/ApplicationDbContext.cs [hunk 13]
```diff
@@ -1,6 +1,12 @@
 ﻿using DAL.Entities;
 using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
```
Comment (romanstat) [inline InternshipBE/DAL/Database/ApplicationDbContext.cs]: remove unused libraries

File: InternshipBE/DAL/Entities/Evaluation.cs [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DAL.Entities
+{
+    public class Evaluation
+    {
+        public int Id { get; set; }
+
+        public int FeedbackId { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Evaluation.cs:9]: check ef core relationship

File: InternshipBE/DAL/Entities/Candidate.cs [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+        public int TestTaskEvaluation { get; set; }
+
+        public Status Status { get; set; }
+
+        public Stack Stack { get; set; }
+
+        public EnglishLevel EnglishLevel { get; set; }  
+
+        public bool ArePlainingToJoin { get; set; }
```
Comment (Amarozka123) [inline InternshipBE/DAL/Entities/Candidate.cs]: Rename to IsPlanningToJoin

File: InternshipBE/DAL/Entities/Skill.cs [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public class Skill
+    {
+        public int Id { get; set; }
+
+        public Stack Stack { get; set; }
+
+        public string Name { get; set; }
+
+        public string IsHardSkill { get; set; }
```
Comment (Amarozka123) [inline InternshipBE/DAL/Entities/Skill.cs]: IsHardSkill  should be bool type

File: InternshipBE/DAL/Entities/User.cs [hunk 17]
```diff
@@ -1,8 +1,24 @@
...
+using System.Threading.Tasks;
 
 namespace DAL.Entities
 {
 	public class User : IdentityUser
 	{
-	}	
+        public int Id { get; set; }
+
+        public int RoleId { get; set; }
```
Comment (Amarozka123) [inline InternshipBE/DAL/Entities/User.cs]: Is user can have only one role?

File: InternshipBE/DAL/Entities/Candidate.cs [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+
+        public string Skype { get; set; }
+
+        public string Education { get; set; }
+
+        public string Links { get; set; }
+
+        public string OtherInfo { get; set; }
+
+        public string PrimarySkill { get; set; }
```
Comment (Amarozka123) [inline InternshipBE/DAL/Entities/Candidate.cs:29]: It should be link to Skills table

File: InternshipBE/DAL/Database/ApplicationDbContext.cs [hunk 19]
```diff
@@ -9,6 +9,23 @@ public class ApplicationDbContext : IdentityDbContext<User>
 		public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options) 
 		{
 		}
+		public DbSet<User> User { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Database/ApplicationDbContext.cs]: Users

File: InternshipBE/DAL/Entities/Candidate.cs [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+        public EnglishLevel EnglishLevel { get; set; } 
+        
+        public bool IsPlanningToJoin { get; set; }
+
+        public DateTime RegistationDate { get; set; }
+
+        public Internship Internship { get; set; }
+
+        public ICollection<UserCandidate> UserCandidate { get; set; }
+
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Candidate.cs]: remove empty line

File: InternshipBE/DAL/Entities/Role.cs [hunk 21]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DAL.Entities
+{
+    public class Role
+    {
+        public int Id { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Role.cs]: remove role

File: InternshipBE/DAL/Entities/User.cs [hunk 22]
```diff
@@ -1,8 +1,23 @@
...
+
+        public string Login { get; set; }
+
+        public string Password { get; set; }
+
+        public string Position { get; set; }
+
+        public DateTime BestContactTime { get; set; }
+
+        public Role Role { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/User.cs]: remove role entity

**Comment by romanstat** [comment]
Remove folder migration and add new after updates

File: InternshipBE/DAL/Entities/Feedback.cs [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+        public UserCandidate UserCandidate { get; set; }
```
Comment (romanstat) [inline InternshipBE/DAL/Entities/Feedback.cs:19]: ICollection

File: InternshipBE/WebApi/Startup.cs [hunk 24]
```diff
@@ -27,12 +27,10 @@ public Startup(IConfiguration configuration)
 		// This method gets called by the runtime. Use this method to add services to the container.
 		public void ConfigureServices(IServiceCollection services)
 		{
-			services.AddDbContext<ApplicationDbContext>(options =>
-				options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
-
 			services.AddIdentity<User, IdentityRole>()
 				.AddEntityFrameworkStores<ApplicationDbContext>();
 
+			services.AddDbContext<ApplicationDbContext>(options => options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection")));
```
Comment (romanstat) [inline InternshipBE/WebApi/Startup.cs]: You added User and IdentityRole to ApplicationDbContext that does not exist and then create ApplicationDbContext, swap

**Review by romanstat — APPROVED** [review]

**Review by Evgeniy-Sinicin — APPROVED** [review]

**Review by Nikityanov — APPROVED** [review]

**Pull request merged**